tests/phpunit/tests/media.php

```diff
@@ -5007,63 +5118,17 @@ public function test_wp_get_loading_optimization_attributes_low_res_image() {
 
 	/**
 	 * @ticket 58681
-	 *
-	 * @dataProvider data_wp_get_loading_optimization_attributes_in_shortcodes
 	 */
-	public function test_wp_get_loading_optimization_attributes_in_shortcodes( $setup, $expected, $message ) {
+	public function test_wp_get_loading_optimization_attributes_in_shortcodes() {
```

Comment:
Rather than removing all of the different use cases that were previously being tested, can we update the data provider to ensure the `do_shortcode` context is performing as expected given the logic changes made in the switch statement in `wp_get_loading_optimization_attributes`?

Reply:
@joemcgill The updated test is effectively the same as the third scenario from the original data provider which IMO is the only scenario still relevant with the new logic. The main query loop consideration, which was what the other two scenarios were covering is no longer relevant now.

The one thing that replaces that logic for which right now there's no concrete test is to cover the early return when using the `do_shortcode` context within e.g. `the_content` filter. I'll add tests for that.

Reply:
Right. It makes sense that the scenarios need to be updated, since some of the originals no longer apply. I just want to make sure we have coverage for all expected use cases.

Reply:
@joemcgill I have added another test with a data provider to cover the `wp_get_loading_optimization_attributes()` changes for early return related to this specifically.